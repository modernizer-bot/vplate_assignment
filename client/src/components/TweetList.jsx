import React from 'react';
import { useSelector } from 'react-redux';
import styled from 'styled-components';

import { selectTweetList } from '../../features/tweet/selectors';
import Tweet from './Tweet';

export default function TweetList() {
  const tweetList = useSelector(selectTweetList);

  return (
    <TweetListInnerBox>
      {tweetList.length === 0 ? (
        <EmptyTweetList>Tweet List is empty :(</EmptyTweetList>
      ) : (
        tweetList.map((tweetInfo) => (
          <Tweet key={tweetInfo.id} tweetInfo={tweetInfo} />
        ))
      )}
    </TweetListInnerBox>
  );
}

const TweetListInnerBox = styled.ul``;

const EmptyTweetList = styled.div``;
