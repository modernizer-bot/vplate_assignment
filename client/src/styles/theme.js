const color = {
  lightGray: '#eff3f4',
  darkGray: '8E99A1',
  twitter: '#31a4f1',
};

const size = {
  profileImageSize: '48px',
  deviceSize: '650px',
  headerSize: '130px',
};

const font = {
  small: '15px',
  medium: '20px',
  big: '30px',
};

const theme = {
  size,
  color,
  font,
};

export default theme;
