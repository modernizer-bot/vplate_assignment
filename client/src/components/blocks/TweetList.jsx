import React from 'react';
import { useSelector } from 'react-redux';
import styled from 'styled-components';

import {
  selectStreamStatus,
  selectTweetList,
} from '../../features/tweet/selectors';

import Tweet from './Tweet';
import TweetLoading from './TweetLoading';

import { center } from '../../styles/mixin';

export default function TweetList() {
  const tweetList = useSelector(selectTweetList);
  const streamStatus = useSelector(selectStreamStatus);

  if (tweetList.length === 0 && !streamStatus) {
    return (
      <EmptyTweetTextWrapper>
        <EmptyTweetText>Tweet List is empty :(</EmptyTweetText>
      </EmptyTweetTextWrapper>
    );
  }

  return (
    <>
      {tweetList.map((tweetInfo) => (
        <Tweet key={tweetInfo.id} tweetInfo={tweetInfo} />
      ))}
      <TweetLoading />
    </>
  );
}

const EmptyTweetTextWrapper = styled.div`
  ${center}
  height: 100%;
  margin-top: 50px;
`;

const EmptyTweetText = styled.p`
  font-size: ${({ theme }) => theme.font.big};
  font-weight: bold;
`;
