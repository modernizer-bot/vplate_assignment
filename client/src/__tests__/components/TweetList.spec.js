import React from 'react';
import { Provider } from 'react-redux';
import { ThemeProvider } from 'styled-components';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';

import theme from '../../styles/theme';
import TweetList from '../../components/blocks/TweetList';

describe('TweetList component test', () => {
  const mockStore = configureStore();

  it('test tweetList feature render ', () => {
    const sampleTweet = {
      author: {
        id: '1383672481316102153',
        name: '일납 고등학교장 오사',
        profile_image_url:
          'https://pbs.twimg.com/profile_images/1458083998815633410/R0wPqkNE_normal.jpg',
        username: 'ASTRO_OHMYGIRL_',
      },
      content: {
        text: 'RT @cat_momococo: 고양이 잠투정 부리는 영상 https://t.co/zy9hrHo9dD',
        mediaUrl:
          'https://pbs.twimg.com/ext_tw_video_thumb/1459327318951096320/pu/img/r7oidDptjJ7jXyKC.jpg',
      },
      id: '1459704809703886848',
      publicMetrics: { retweet_count: 0, reply_count: 0, like_count: 0 },
    };

    const initialState = {
      tweet: {
        tweetList: [sampleTweet],
        streamStatus: false,
      },
    };

    const store = mockStore(initialState);

    store.dispatch = jest.fn();

    const tree = renderer
      .create(
        <Provider store={store}>
          <ThemeProvider theme={theme}>
            <TweetList />
          </ThemeProvider>
        </Provider>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
