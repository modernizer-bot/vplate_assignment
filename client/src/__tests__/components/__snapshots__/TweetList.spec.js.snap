// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TweetList test Tweet & TweetLoading render test 1`] = `
Array [
  <li
    className="sc-fFeiMQ eUeroC"
  >
    <a
      className="sc-bkkeKt ciBhox"
      href="https://twitter.com/ASTRO_OHMYGIRL_/status/1459704809703886848"
    >
      <div
        className="sc-bdvvtL ejCBZN"
      >
        <div
          className="sc-gsDKAQ dJUObk"
        >
          <div
            className="sc-jRQBWg fhsQcj"
          >
            <img
              alt="author-profile"
              className="sc-gKclnd bxIsmi"
              src="https://pbs.twimg.com/profile_images/1458083998815633410/R0wPqkNE_normal.jpg"
            />
          </div>
          <div
            className="sc-dkPtRN jrokNO"
          >
            <div
              className="sc-hKwDye hXCNow"
            >
              일납 고등학교장 오사
            </div>
            <div
              className="sc-eCImPb cA-DrhV"
            >
              @ASTRO_OHMYGIRL_
            </div>
          </div>
        </div>
      </div>
      <div
        className="sc-kDTinF gfaHiN"
      >
        <div
          className="sc-jrQzAO dJFCvp"
        >
          <p
            className="sc-pVTFL eYLufl"
          >
            RT @cat_momococo: 고양이 잠투정 부리는 영상 https://t.co/zy9hrHo9dD
          </p>
        </div>
        <div
          className="sc-iCfMLu GTtny"
        >
          <img
            alt="media-content"
            className="sc-furwcr fXzvMw"
            src="https://pbs.twimg.com/ext_tw_video_thumb/1459327318951096320/pu/img/r7oidDptjJ7jXyKC.jpg"
          />
        </div>
      </div>
      <div
        className="sc-hBUSln lccWIB"
      >
        <div
          className="sc-fotOHu fYYYRR"
        >
          <div
            className="sc-egiyK rWYOo"
          >
            <div
              className="sc-iqseJM qZKvI"
            >
              <div
                className="sc-crHmcD eYzDfU"
              >
                <svg
                  fill="currentColor"
                  height={24}
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": undefined,
                    }
                  }
                  viewBox="0 0 24 24"
                  width={24}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12 2C6.486 2 2 5.589 2 10c0 2.908 1.898 5.516 5 6.934V22l5.34-4.005C17.697 17.852 22 14.32 22 10c0-4.411-4.486-8-10-8zm0 14h-.333L9 18v-2.417l-.641-.247C5.67 14.301 4 12.256 4 10c0-3.309 3.589-6 8-6s8 2.691 8 6-3.589 6-8 6z"
                  />
                </svg>
              </div>
            </div>
            <div
              className="sc-bqiRlB jmJzvV"
            >
              <div
                className="sc-ksdxgE ByDGA"
              >
                0
              </div>
            </div>
          </div>
          <div
            className="sc-egiyK rWYOo"
          >
            <div
              className="sc-iqseJM qZKvI"
            >
              <div
                className="sc-crHmcD eYzDfU"
              >
                <svg
                  fill="currentColor"
                  height={24}
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": undefined,
                    }
                  }
                  viewBox="0 0 1024 1024"
                  width={24}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M136 552h63.6c4.4 0 8-3.6 8-8V288.7h528.6v72.6c0 1.9.6 3.7 1.8 5.2a8.3 8.3 0 0 0 11.7 1.4L893 255.4c4.3-5 3.6-10.3 0-13.2L749.7 129.8a8.22 8.22 0 0 0-5.2-1.8c-4.6 0-8.4 3.8-8.4 8.4V209H199.7c-39.5 0-71.7 32.2-71.7 71.8V544c0 4.4 3.6 8 8 8zm752-80h-63.6c-4.4 0-8 3.6-8 8v255.3H287.8v-72.6c0-1.9-.6-3.7-1.8-5.2a8.3 8.3 0 0 0-11.7-1.4L131 768.6c-4.3 5-3.6 10.3 0 13.2l143.3 112.4c1.5 1.2 3.3 1.8 5.2 1.8 4.6 0 8.4-3.8 8.4-8.4V815h536.6c39.5 0 71.7-32.2 71.7-71.8V480c-.2-4.4-3.8-8-8.2-8z"
                  />
                </svg>
              </div>
            </div>
            <div
              className="sc-bqiRlB jmJzvV"
            >
              <div
                className="sc-ksdxgE ByDGA"
              >
                0
              </div>
            </div>
          </div>
          <div
            className="sc-egiyK rWYOo"
          >
            <div
              className="sc-iqseJM qZKvI"
            >
              <div
                className="sc-crHmcD eYzDfU"
              >
                <svg
                  fill="currentColor"
                  height={24}
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": undefined,
                    }
                  }
                  viewBox="0 0 1024 1024"
                  width={24}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M923 283.6a260.04 260.04 0 0 0-56.9-82.8 264.4 264.4 0 0 0-84-55.5A265.34 265.34 0 0 0 679.7 125c-49.3 0-97.4 13.5-139.2 39-10 6.1-19.5 12.8-28.5 20.1-9-7.3-18.5-14-28.5-20.1-41.8-25.5-89.9-39-139.2-39-35.5 0-69.9 6.8-102.4 20.3-31.4 13-59.7 31.7-84 55.5a258.44 258.44 0 0 0-56.9 82.8c-13.9 32.3-21 66.6-21 101.9 0 33.3 6.8 68 20.3 103.3 11.3 29.5 27.5 60.1 48.2 91 32.8 48.9 77.9 99.9 133.9 151.6 92.8 85.7 184.7 144.9 188.6 147.3l23.7 15.2c10.5 6.7 24 6.7 34.5 0l23.7-15.2c3.9-2.5 95.7-61.6 188.6-147.3 56-51.7 101.1-102.7 133.9-151.6 20.7-30.9 37-61.5 48.2-91 13.5-35.3 20.3-70 20.3-103.3.1-35.3-7-69.6-20.9-101.9zM512 814.8S156 586.7 156 385.5C156 283.6 240.3 201 344.3 201c73.1 0 136.5 40.8 167.7 100.4C543.2 241.8 606.6 201 679.7 201c104 0 188.3 82.6 188.3 184.5 0 201.2-356 429.3-356 429.3z"
                  />
                </svg>
              </div>
            </div>
            <div
              className="sc-bqiRlB jmJzvV"
            >
              <div
                className="sc-ksdxgE ByDGA"
              >
                0
              </div>
            </div>
          </div>
          <div
            className="sc-iqseJM qZKvI"
          >
            <div
              className="sc-crHmcD eYzDfU"
            >
              <svg
                fill="currentColor"
                height={24}
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 1024 1024"
                width={24}
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M400 317.7h73.9V656c0 4.4 3.6 8 8 8h60c4.4 0 8-3.6 8-8V317.7H624c6.7 0 10.4-7.7 6.3-12.9L518.3 163a8 8 0 0 0-12.6 0l-112 141.7c-4.1 5.3-.4 13 6.3 13zM878 626h-60c-4.4 0-8 3.6-8 8v154H214V634c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8v198c0 17.7 14.3 32 32 32h684c17.7 0 32-14.3 32-32V634c0-4.4-3.6-8-8-8z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </a>
  </li>,
  <div
    className="sc-ieecCq hvboxu"
  >
    <p
      className="sc-dJjYzT jzKXDy"
    >
      Loading
    </p>
  </div>,
]
`;
